### PR TITLE
Added match draw feature

### PR DESCRIPTION
**Pull request by Ksathwik03**

### CODE REVIEW

File: cogs/matches.py [hunk 1]
```diff
@@ -211,6 +210,25 @@ async def invalidate(self, ctx):
...
+        if not self.db.in_a_match(ctx.guild.id, ctx.author.id):
+            await discord_.send_message(ctx, f"User {ctx.author.mention} is not in a match.")
+            return
+        match_info = self.db.get_match_info(ctx.guild.id, ctx.author.id)
+        opponent = await discord_.fetch_member(ctx.guild, match_info.p1_id if match_info.p1_id != ctx.author.id else match_info.p2_id)
+        await ctx.send(f"{opponent.mention} you opponent {ctx.author.mention} has proposed to draw the match, type `yes` within 30 seconds to accept")
+
+        try:
+            message = await self.client.wait_for('message', timeout=30, check=lambda message: message.author == opponent and message.content.lower() == 'yes')
+            self.db.delete_match(ctx.guild.id, ctx.author.id)
```
Comment (pseudocoder10) [inline cogs/matches.py:225]: We don't want to just delete the match, the match should be added in the list of finished matches (with final scores 0-0) and the ratings of both the participants should be updated.

File: cogs/matches.py [hunk 2]
```diff
@@ -211,6 +210,25 @@ async def invalidate(self, ctx):
...
+            await ctx.send(embed=discord.Embed(description=f"{ctx.author.mention} {opponent.mention}, match has been draw", color=discord.Color.green()))
```
Comment (pseudocoder10) [inline cogs/matches.py:226]: The bot should send the final result/rating changes in that channel just like any other match.

File: cogs/matches.py [hunk 3]
```diff
@@ -211,6 +210,25 @@ async def invalidate(self, ctx):
...
+        except asyncio.TimeoutError:
+            await ctx.send(f"{ctx.author.mention} your opponent didn't respond in time")
+
+      
+
```
Comment (pseudocoder10) [inline cogs/matches.py:231]: Remove extra spacing.

**Review by pseudocoder10 — CHANGES_REQUESTED** [review]

File: cogs/matches.py [hunk 4]
```diff
@@ -223,12 +223,23 @@ async def draw(self,ctx):
...
-            await ctx.send(embed=discord.Embed(description=f"{ctx.author.mention} {opponent.mention}, match has been draw", color=discord.Color.green()))
+            self.db.add_to_finished(match_info, "draw")
+            channel = self.client.get_channel(match_info.channel)
+            embed = discord.Embed(color=discord.Color.green())
+            pos, name, ratingChange = '', '', ''
+            players = [ctx.author,opponent]
+            for player in players:
+                pos += f"{':first_place:'}\n"
+                name += f"{player.mention}\n"
+                ratingChange += f"{self.db.get_match_rating(ctx.guild.id, player.id)[-1]}+0\n"
```
Comment (pseudocoder10) [inline cogs/matches.py:234]: Why is the rating change = 0 ? Even if the match is drawn, there will be some rating change.

Preferably can you make the implementation similar to [this](https://github.com/pseudocoder10/Lockout-Bot/blob/master/cogs/matches.py#L352-L375) ? Only change will be that the scores will be 0-0.

**Review by pseudocoder10 — CHANGES_REQUESTED** [review]

**Review by pseudocoder10 — CHANGES_REQUESTED** [review]
Can you make a separate issue and pr for this? This seems unrelated to #3.

**Comment by Ksathwik03** [comment]
Sure sir , is the match draw feature ok.

File: cogs/matches.py (hunk 4 above)
Comment (Ksathwik03) [reply cogs/matches.py:234]: Yes sir i will make the change and sure I will make a seperate pr for duplicate handles

**Pull request closed without merging**